lms/djangoapps/ccx/utils.py

```diff
@@ -334,3 +339,93 @@ def get_course_chapters(course_key):
         return course_struct['blocks'][course_struct['root']].get('children', [])
     except KeyError:
         return []
+
+
+def add_master_course_staff_to_ccx(master_course, ccx_key, display_name, send_email=True):
+    """
+    Add staff and instructor roles on ccx to all the staff and instructors members of master course.
+
+    Arguments:
+        master_course (CourseDescriptorWithMixins): Master course instance
+        ccx_key (CCXLocator): CCX course key
+        display_name (str): ccx display name for email
+    """
+    list_staff = list_with_level(master_course, 'staff')
+    list_instructor = list_with_level(master_course, 'instructor')
+
+    with ccx_course(ccx_key) as course_ccx:
+        email_params = get_email_params(course_ccx, auto_enroll=True, course_key=ccx_key, display_name=display_name)
+        list_staff_ccx = list_with_level(course_ccx, 'staff')
+        list_instructor_ccx = list_with_level(course_ccx, 'instructor')
+        for staff in list_staff:
+            # this call should be idempotent
+            if staff not in list_staff_ccx:
+                # allow 'staff' access on ccx to staff of master course
+                allow_access(course_ccx, staff, 'staff')
+
+                # Enroll the staff in the ccx
+                enroll_email(
+                    course_id=ccx_key,
+                    student_email=staff.email,
+                    auto_enroll=True,
+                    email_students=send_email,
+                    email_params=email_params,
+                )
+
+        for instructor in list_instructor:
+            # this call should be idempotent
+            if instructor not in list_instructor_ccx:
+                # allow 'instructor' access on ccx to instructor of master course
+                allow_access(course_ccx, instructor, 'instructor')
+
+                # Enroll the instructor in the ccx
+                enroll_email(
+                    course_id=ccx_key,
+                    student_email=instructor.email,
+                    auto_enroll=True,
+                    email_students=send_email,
+                    email_params=email_params,
+                )
```

Comment:
Moving my comment from the migration module to the implementation here:

My concern is that if you don't wrap these enroll_email calls in a try/except block you could end in an unexpected state. enroll_email can throw: CourseEnrollmentException as well as smtplib.SMTPException from django. What should the behavior here be when these exceptions are thrown?